examples/service/random.toit

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2023 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the examples/LICENSE file.
+
+// This example is the companion to https://github.com/toitlang/toit/discussions/869
+// with the full working source code for the presented snippets.
+
+import log
+import monitor
+import system.services show ServiceClient ServiceSelector ServiceResourceProxy
+import system.services show ServiceProvider ServiceHandler ServiceResource
+
+interface RandomService:
+  static SELECTOR ::= ServiceSelector
+      --uuid="dd9e5fd1-a5e9-464e-b2ef-92bf15ea02ca"
+      --major=0
+      --minor=1
+
+  next limit/int -> int
+  static NEXT_INDEX /int ::= 0
+
+  create_die sides/int -> int
+  static CREATE_DIE_INDEX /int ::= 1
+
+  roll_die handle/int -> int
+  static ROLL_DIE_INDEX /int ::= 2
+
+main:
+  spawn::
+    service := RandomServiceProvider
+    service.install
+    service.uninstall --wait
+
+  client := RandomServiceClient
+  client.open
+  10.repeat:
+    print "random = $(client.next 100)"
+
+  die := client.create_die 6
+  10.repeat:
+    print "die roll = $(die.roll)"
+  sleep (Duration --s=10)
+  die.close
+
+class RandomServiceClient extends ServiceClient implements RandomService:
+  static SELECTOR ::= RandomService.SELECTOR
+  constructor selector/ServiceSelector=SELECTOR:
+    assert: selector.matches SELECTOR
+    super selector
+
+  next limit/int -> int:
+    return invoke_ RandomService.NEXT_INDEX limit
```

Comment:
Does the example show us using a private method that might not be available if we tighten up the rules on privacy?

Reply:
Yes. There are 1^20 occurrences of this.